src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -400,12 +403,19 @@ internal async Task SendInvocationRequest(ScriptInvocationContext context)
         {
             try
             {
+                // do not send invocation requests for functions that failed to load or for functions that could not be indexed by the worker
                 if (_functionLoadErrors.ContainsKey(context.FunctionMetadata.GetFunctionId()))
                 {
                     _workerChannelLogger.LogDebug($"Function {context.FunctionMetadata.Name} failed to load");
                     context.ResultSource.TrySetException(_functionLoadErrors[context.FunctionMetadata.GetFunctionId()]);
                     _executingInvocations.TryRemove(context.ExecutionContext.InvocationId.ToString(), out ScriptInvocationContext _);
                 }
+                else if (_metadataRequestErrors.ContainsKey(context.FunctionMetadata.GetFunctionId()))
+                {
+                    _workerChannelLogger.LogDebug($"Function {context.FunctionMetadata.Name} failed to load");
+                    context.ResultSource.TrySetException(new Exception(_metadataRequestErrors[context.FunctionMetadata.GetFunctionId()].Message));
```

Comment:
You shouldn't have to create a new `Exception` here -- can you just do `TrySetException(_metadataRequestErrors[context.FunctionMetadata.GetFunctionId()])`?

Reply:
The new `Exception` is needed unfortunately because `_metadataRequestErrors[context.FunctionMetadata.GetFunctionId()]` gives an `RpcException` while `context.ResultSource.TrySetException` requires something of type `Exception`.

Reply:
Can you see how `_functionLoadErrors` handles this? I think that it may be changing that RpcException into an actual Exception before putting it in the collection. This should do the same thing, if possible.

Reply:
Good catch, turns out there is a method that checks the status and returns an `RpcException `(different `RpcException `than what I was defining). 

Making the change now!